### PR TITLE
Support Parsing `ALTER TYPE` in openGauss

### PR DESCRIPTION
**Pull request by everly-gif**
Ref #17736 

Changes proposed in this pull request:
- Proofreads `ALTER TYPE` grammar
- Supports Parsing `ALTER TYPE` in openGauss
- Adds tests

✅ Builds locally

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/DDLStatement.g4 [hunk 1]
```diff
@@ -1161,7 +1161,6 @@ alterTypeClauses
     | RENAME TO name
     | RENAME ATTRIBUTE name TO name dropBehavior?
     | SET SCHEMA name
-    | SET LP_ operatorDefList RP_
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/DDLStatement.g4:1164]: Does openGauss not support this syntax?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/AlterTypeStatement.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import lombok.ToString;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
+
+/**
+ * Alter type statement.
+ */
+@ToString
+public abstract class AlterTypeStatement extends AbstractSQLStatement implements DDLStatement {
+
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/AlterTypeStatement.java:28]: Please remove this useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/DDLStatement.g4 (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/DDLStatement.g4:1164]: Yes, It isnt mentioned in their syntax for `ALTER TYPE`

You can see it here : https://opengauss.org/en/docs/3.0.0/docs/Developerguide/alter-type.html

Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/DDLStatement.g4:1164]: > Yes, It isnt mentioned in their syntax for `ALTER TYPE`
> 
> You can see it here : https://opengauss.org/en/docs/3.0.0/docs/Developerguide/alter-type.html

Ok, I got it.

**Review by strongduanmu — APPROVED** [review]
Thank you @everly-gif, another perfect pr.

**Pull request merged**